libcontainer/rootfs_linux.go

```diff
@@ -1037,28 +1037,28 @@ func writeSystemProperty(key, value string) error {
 }
 
 func remount(m *configs.Mount, rootfs string, mountFd *int) error {
-	source := m.Source
+	srcFD := ""
 	if mountFd != nil {
-		source = "/proc/self/fd/" + strconv.Itoa(*mountFd)
+		srcFD = "/proc/self/fd/" + strconv.Itoa(*mountFd)
 	}
 
-	return utils.WithProcfd(rootfs, m.Destination, func(procfd string) error {
+	return utils.WithProcfd(rootfs, m.Destination, func(dstFD string) error {
 		flags := uintptr(m.Flags | unix.MS_REMOUNT)
-		err := mount(source, m.Destination, procfd, m.Device, flags, "")
+		err := mountViaFDs(m.Source, srcFD, m.Destination, dstFD, m.Device, flags, "")
 		if err == nil {
 			return nil
 		}
 		// Check if the source has ro flag...
 		var s unix.Statfs_t
-		if err := unix.Statfs(source, &s); err != nil {
-			return &os.PathError{Op: "statfs", Path: source, Err: err}
+		if err := unix.Statfs(m.Source, &s); err != nil {
```

Comment:
path argument of `unix.Statfs` used to be proc fd path if `mountFd != nil`, is this gonna be a problem?

Reply:
Good question. I think it might be. Let me fix this.

Reply:
OK I was not able to repro the error that might happen here (as this requires a very narrow scenario that uses a bind mount of a read-only fuse-sshfs mount), but I fixed the issue so the statfs() uses either source or srcFD.

PTAL @hqhq